### DISCUSSION TITLE
Allow partial credit for each day

### DISCUSSION THREAD

**Discussion by rplazevedo** [iSoron/uhabits#1267] [Feature Requests]
I recently watched a [video by CGP Grey](https://youtu.be/fSwpe8r50_o?t=104), and the key idea that stuck with me was a partial credit system in the habit tracker. This could be implemented by double clicking the tracker for example (in the widget, 1 click for full credit, as normal, 2 clicks to switch to partial, 3 clicks to reset). They could count a custom or fixed percentage of the usual score (doesn't really matter to me, anywhere from 20% to 50% would suit me just fine). Personally, I think this would fit some of my habits very well and be really useful.

**Comment by iSoron** [comment]
Thanks for the feature request, @rplazevedo. Note that if you create measurable habits, Loop already awards you partial credits. For example, if your habit is "run 10 miles per day", but you only run 5 miles, your score should eventually settle at 50%.

**Reply by rplazevedo** [reply]
I think there are two problems with using measurable habits for my purpose. One of which is that there is no widget to track a measurable habits, which is what I use to avoid opening the app every time. The other is that the partial success might not scale linearly with the the total goal. Say the habit I want to track is "Exercise". I might want to give full credit for any sessions over 30 minutes, but want to set up a partial credit of 50% for anytime that I work out for even just five minutes. In this scenario, what's important to me is not "minutes spent working out", it's the maintenance of the habit itself.

**Reply by iSoron** [reply]
> there is no widget to track a measurable habits

You should be able to create widgets for measurable habits. If you are not able to, that's a bug. Please feel free to submit a bug report. The only exception are streaks, since we have not implemented streaks for measurable habits yet.

> I might want to give full credit for any sessions over 30 minutes, but want to set up a partial credit of 50% for anytime that I work out for even just five minutes.

That's an interesting suggestion. If anyone else would like this feature, please feel free to upvote this discussion and leave a comment. We may revisit it in the future.

**Reply by herrbrixter** [reply]
I agree with @iSoron. Just create a measurable habit like 1 to 3. 1 - failed, 2 - partial, 3 - completed. There is no need to modify Yes or No habits.

**Comment by lily-peng** [comment]
I'm interested in something similar. I think this goes back to a comment @iSoron wrote on Reddit about "categorical habits" https://www.reddit.com/r/getdisciplined/comments/8mty7l/z/e0ytg97

In my use case, I think a yes/no is a little strict with calorie-counting. Ideally, I'd like at least 1 middle stage, like a partial credit, if I ate over my limit but under another limit I'd consider "too much".

The "at most" feature may help a little in my case, but it isn't in the app yet (any word on when 2.1 will release?)

**Comment by jasperandrew** [comment]
I'm also interested in something like this, as illustrated in the video @rplazevedo linked to. I can kind of hack this feature by using the "?" tri-state mode, but it would be nice to maintain the ? functionality as well. I'm thinking about trying to implement something like this, maybe just a quad-state mode, if I find the time. Doesn't seem too complex, at least from the outside without looking at code lol.

**Comment by neube3** [comment]
Didn't watch the video yet, but I've heard an argumentation about allowing partial successes removes the threat of "oh, well" effect. Illustrative example:
Someone has a yes/no habit "I didn't smoke today". If they break the habit by smoking a cigarette, there is the "oh, well" effect - "since I broke the streak/habit, I might as well smoke two packs". Partial goals ("smoke less than 3 cigarettes") are supposed to be the solution - as one would have another stop point to strive for.
